docs/en/sql-reference/statements/select/index.md

```diff
@@ -265,5 +266,19 @@ SELECT * REPLACE(i + 1 AS i) EXCEPT (j) APPLY(sum) from columns_transformers;
 └─────────────────┴────────┘
 ```
 
+## SETTINGS in SELECT Query {#settings-in-select}
+
+You can specify the necessary settings right in the `SELECT` query. The setting value is applied only to this query and is reset to default or previous value after the query is executed. 
+
+To specify several settings, use several `SETTINGS` clauses. 
+
+Other ways to make settings, see [here](../../../operations/settings/index.md). 
+
+**Example**
+
+``` sql
+SELECT * FROM some_table SETTINGS optimize_read_in_order=1 SETTINGS cast_keep_nullable=1;
```

Comment:
```suggestion
SELECT * FROM some_table SETTINGS optimize_read_in_order=1, cast_keep_nullable=1;
```

Multiple settings are separated by a comma. `SELECT * FROM some_table SETTINGS optimize_read_in_order=1 SETTINGS cast_keep_nullable=1;` works too (a bit different way, see a note below), but if you try to specify more than two `SETTINGS` clauses it will fail with syntax error:
```
:) SELECT * FROM some_table SETTINGS optimize_read_in_order=1 SETTINGS cast_keep_nullable=1 SETTINGS max_threads=8

Syntax error: failed at position 84 ('SETTINGS'):

select * from mydb SETTINGS optimize_read_in_order=1 SETTINGS cast_keep_nullable=1 SETTINGS max_threads=8

Expected one of: number, literal, NULL, FALSE, TRUE
```

It happens because of the way that query parsing implemented. You can specify `SETTINGS` clause for every simple `SELECT` subquery of some complex `SELECT` query and for the whole query, e.g. `(SELECT ... SETTINGS ...) UNION ALL (SELECT ... SETTINGS ...) SETTINGS ...`. In this example the first `SETTINGS` clause should be applied to the first subquery, the second `SETTINGS` clause should be applied to the second subquery and the third clause should be applied to the whole query.

However, we have some issues with subquery settings (see #11804 comments and related issues for example), so I'm not sure if we should describe this "feature" in documentation or not. Probably we should describe only top-level `SETTINGS` clause (the one which works for the whole query) or add a note that some subquery settings may work unexpected way.

Reply:
We should not describe the hack with two SETTINGS clauses in docs.

Reply:
@tavplubix @alexey-milovidov Thanks for the review! Fixed the description and the example.